_ap/17/index.adoc

```diff
@@ -0,0 +1,62 @@
+---
+num: 17
+tags:
+  - "tls"
+  - "certificates"
+  - "dns"
+  - "Letsencrypt"
+title: "Automatic management of certificates per data plane cluster using Letsencrypt"
+status: "Accepted"
+authors:
+  - "Manyanda Chitimbo"
+---
+
+## Intent
+
+Define a pattern for the automatic management of certificates per data plane cluster using Letsencrypt
+
+## Motivation
+
+When a managed service is deployed on the data plane cluster its routes have be reachable via TLS.
+
+It means that a managed service a way to the clients to verify and validate the TLS connection with the related handshake with the service exposed outside of the data plane cluster.
+
+## Applicability
+
+This pattern applies to any service that wants a managed service routes to be reachable via TLS
+
+## Description
+
+A managed service will use https://Letsencrypt.org[Letsencrypt] to managed certificates on the fly i.e automatic and this removing the need for manual intervention and the toil related to this.
+
+To avoid sharing of certificates between data planes, managed service implementors are recommended to generate a wildcard certificate per data plane.
+
+As an example, if the root domain is `myservice.bf2.dev`, then a wildcard certificate will be generated for each data plane cluster under a subdomain `<cluster-id>.myservice.bf2.dev`.
+
+Furthermore, all routes for the managed services deployed on the the data plane will be of the form `foobar.<cluster-id>.myservice.bf2.dev`.
+
+NOTE: There is a more granular approach to avoid certificate sharing by generating a widlcard certificate per deployed service, however this approach is likely to encounter and hits https://Letsencrypt.org/docs/rate-limits/[Letsencrypt limits] more frequently. 
```

Comment:
```suggestion
NOTE: There is a more granular approach to avoid certificate sharing by generating a widlcard certificate per deployed service instance, however this approach is likely to encounter and hits https://Letsencrypt.org/docs/rate-limits/[Letsencrypt limits] more frequently. 
```